libsupport/include/katana/Result.h

```diff
@@ -266,8 +266,15 @@ class KATANA_EXPORT [[nodiscard]] ErrorInfo {
 
     fmt::format_to(std::back_inserter(out), " ({}:{})", base_name, line_no);
 
+    std::string code = ec.message();
+    std::string msg(out.data());
+
+    std::string combined = "\n" + code + "\n" + msg;
+    const char* output = combined.c_str();
+
     ErrorInfo ei(ec);
-    ei.Prepend(out.data(), out.data() + out.size());
+    // TODO(micah) sometimes random characters get printed at the end.
```

Comment:
> sometimes random characters get printed at the end.

This seems problematic. We should probably understand why before inflicting this change to everyone.

At a higher level, if someone needs a specific error output, they should just write the output they want rather than changing the format for everyone. Generally, the message output is arbitrary and subject to change.

Reply:
Okay, I was having a hard time figuring out why, but I'll keep looking into it. It's only up to 4 characters at the very end of the message so it's not a huge deal.

Sure, but I need the TCK and QueryErrorCode to print for every error reported, and making a new macro or expecting everyone to remember to do that seems unnecessary.

Reply:
Actually it seems the random characters are causing some python tests to fail so I guess I have to fix it

Reply:
Some string copy function copy in 32-bit chunks and even over copy by as much as 3 bytes. Is one of the inputs not null terminated, so when you over copy the junk at the end is included in the string?

Reply:
Actually I'm pretty sure null termination is the issue. c_str( ) adds a null terminator, but length does not include that null, so when you are prepending below, you are dropping the null which might be messing things up.

Reply:
^ I was thinking something like that was the problem. I added a null terminator and it seems to have fixed it. I really don't know how this wasn't a problem before...

Reply:
Actually I do know why. Didn't convert to a C++ string before.

Reply:
> Sure, but I need the TCK and QueryErrorCode to print for every error reported, and making a new macro or expecting everyone to remember to do that seems unnecessary.

What is wrong with making whatever is actually printing the error message to call something like `FormatForTCK(err)`?

Reply:
Well by the time that it gets to that point the string is the full call stack, extracting that is messier than just adding it in the beginning. If you want me to make it print the error code only when it's a `QueryErrorCode`, I could do that. I do want both errors to print though, because the TCK errors can be unhelpful and I want people to be familiar with the new error codes so they can use them properly and understand them better when they get thrown.